lsst/ts/cbp/statemachine.py

```diff
@@ -0,0 +1,527 @@
+        """
+
+        Parameters
+        ----------
+        id_data
+
+        Returns
+        -------
+        None
+
+        """
+        pass #TODO: finish enterControl
+
+    async def begin_enable(self, id_data):
+        """Overrides the begin_enable function in salobj.BaseCsc to make sure the CBP is unparked.
+
+        Parameters
+        ----------
+        id_data
+
+        Returns
+        -------
+        None
+
+        """
+        self.model._cbp.set_park()
+
+
+class CBPRemote:
+    """This the class that implements the remote functionality for the CBP CSC.
+
+    Attributes
+    ----------
+    remote: salobj.Remote
+        The salobj remote that is used to send commands and receive events and telemetry.
+
+    log: logging.Logger
+        This is the log for the current class.
+
+    """
+    def __init__(self):
+        self.remote = salobj.Remote(SALPY_CBP)
+        self.log = logging.getLogger(__name__)
+
+    async def standby(self):
+        """Calls and awaits the standby command. Logs the acknowledgement code from the command.
+
+        Returns
+        -------
+        None
+
+        """
+        standby_topic = self.remote.cmd_standby.DataType()
+        standby_ack = await self.remote.cmd_standby.start(standby_topic,timeout=10)
+        self.log.info(standby_ack.ack.ack)
+
+    async def disable(self):
+        """Calls and awaits the disable command and logs the acknowledgement code.
+
+        Returns
+        -------
+        None
+
+        """
+        disable_topic = self.remote.cmd_disable.DataType()
+        disable_ack = await self.remote.cmd_disable.start(disable_topic,timeout=10)
+        self.log.info(disable_ack.ack.ack)
+
+    async def start(self):
+        """Calls the start and awaits the result and logs the acknowledgement code.
+
+        Returns
+        -------
+        None
+
+        """
+        start_topic = self.remote.cmd_start.DataType()
+        start_ack = await self.remote.cmd_start.start(start_topic,timeout=10)
+        self.log.info(start_ack.ack.ack)
+
+    async def enable(self):
+        """Calls the enable command and awaits the result which logs the acknowledgement code.
+
+        Returns
+        -------
+        None
+
+        """
+        enable_topic = self.remote.cmd_enable.DataType()
+        enable_ack = await self.remote.cmd_enable.start(enable_topic,timeout=10)
+        self.log.info(enable_ack.ack.ack)
+
+    async def move_azimuth(self, azimuth):
+        """Calls the moveAzimuth command and awaits the result which logs the acknowledgement code.
+
+        Parameters
+        ----------
+        azimuth: float
+            The value of the azimuth to move to in degrees.
+
+        Returns
+        -------
+        None
+
+        """
+        move_azimuth_topic = self.remote.cmd_moveAzimuth.DataType()
+        move_azimuth_topic.azimuth = azimuth
+        move_azimuth_ack = await self.remote.cmd_moveAzimuth.start(move_azimuth_topic,timeout=10)
```

Comment:
All of these commands can only be run in series (or sequentially) as written like this. Is that intentional?
The API supports moving multiple axes simultaneously, no?

Reply:
At this point, the remote class is more of an example to help show a particular way to write it. Essentially, one can write their own remote structure if they want. But I can work on improvements to this example remote. The underlying API handles moving multiple axes just fine. The CSC can also do this, I just have to write the remote calls more asynchronously.